File: altoq_backend/app/schemas/order.py
from pydantic import BaseModel
from datetime import datetime
from typing import List, Optional

class OrderItem(BaseModel):
    productId: int
    quantity: int
    price: float

class OrderBase(BaseModel):
    user_id: int
    products: List[OrderItem]
    total_amount: float
    status: str = "pending"
    shipping_address: str = ""
    contact_phone: str = ""

class OrderCreate(OrderBase):
    pass

class OrderResponse(OrderBase):
    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    delivery_code: Optional[str] = None
    client_name: Optional[str] = None
    client_email: Optional[str] = None

    class Config:
        from_attributes = True

class Order(OrderBase):
    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    delivery_code: Optional[str] = None
    client_name: Optional[str] = None
    client_email: Optional[str] = None

    class Config:
        from_attributes = True

File: altoq_frontend/src/app/models/order.ts
export interface Order {
  id?: number;
  user_id: number;
  products: OrderItem[];
  total_amount: number;
  status: 'pending' | 'confirmed' | 'completed' | 'canceled';
  shipping_address: string;
  contact_phone: string;
  delivery_code?: string;
  created_at?: Date;
  updated_at?: Date;
}

export interface OrderItem {
  productId: number;
  quantity: number;
  price: number;
}